crates/cli/src/cli_args/add.rs

```diff
@@ -0,0 +1,180 @@
+use crate::State;
+use clap::Args;
+use miette::Context;
+use pacquet_package_json::DependencyGroup;
+use pacquet_package_manager::Add;
+use std::path::PathBuf;
+
+#[derive(Debug, Args)]
+pub struct AddDependencyOptions {
+    /// Install the specified packages as regular dependencies.
+    #[clap(short = 'P', long)]
+    save_prod: bool,
+    /// Install the specified packages as devDependencies.
+    #[clap(short = 'D', long)]
+    save_dev: bool,
+    /// Install the specified packages as optionalDependencies.
+    #[clap(short = 'O', long)]
+    save_optional: bool,
+    /// Using --save-peer will add one or more packages to peerDependencies and install them as dev dependencies
+    #[clap(long)]
+    save_peer: bool,
+}
+
+impl AddDependencyOptions {
+    /// Convert the `--save-*` flags to an iterator of [`DependencyGroup`]
+    /// which selects which target group to save to.
+    fn dependency_groups(&self) -> impl Iterator<Item = DependencyGroup> {
+        let &AddDependencyOptions { save_prod, save_dev, save_optional, save_peer } = self;
+        let has_prod = save_prod || (!save_dev && !save_optional && !save_peer); // no --save-* flags implies --save-prod
+        let has_dev = save_dev || (!save_prod && !save_optional && save_peer); // --save-peer with nothing else implies --save-dev
+        let has_optional = save_optional;
+        let has_peer = save_peer;
```

Comment:
why are these new variable needed?

Reply:
should be resolved by https://github.com/pnpm/pacquet/pull/143/commits/11924f4656387898698f48ab801f7dfa128f67af